### DISCUSSION TITLE
Strange output from the CoreML model converted from the Pytorch model

### DISCUSSION THREAD

**Discussion by zhengliu0829** [ultralytics/yolov5#2286] [Q&A] [answered]
Hi,
I trained a YOLOV5S model to detect one object. I converted the model to TensorFlow, TensorFlow lite and CoreML formats using the `export.py` code. The TensorFlow and TensorFlow lite model work well. But the output from the CoreML model is wrong. See the screenshot below. Can anybody please give me some clue why the output is like this?  Is it wrong? Or extra postprocessing steps are needed before feeding this output to the NMS function?

Thanks!

![image](https://user-images.githubusercontent.com/38077556/109014149-dddc6780-7681-11eb-967e-6a9bc75ab935.png)

**Comment by glenn-jocher** [comment, marked as answer]
@zhengliu0829 yes this model would require additional steps from the Detect() layer as well as NMS to run inference.

**Reply by zhengliu0829** [reply]
Thanks @glenn-jocher. Do you know why the Detect() layer is not exported when converting from Pytorm model to CoreML? Is it because the Pytorch "trace" can not trace the for-loop for the bounding boxes calculation? If I convert the model from TensorFlow .pb format to CoreML, will the Detect() layer be available in the CoreML model?
